contracts/Governance.sol

```diff
@@ -0,0 +1,271 @@
+import { MiniMeTokenERC20 as Pinakion } from "kleros-interaction/contracts/standard/arbitration/ArbitrableTokens/MiniMeTokenERC20.sol";
+
+contract Governance {
+
+    Pinakion public pinakion;
+    TokenController public tokenController;
+    ArbitrablePermissionList public proposalList;
+
+    uint public proposalQuorum;
+    uint public currentProposalQuorum;
+
+    uint public quorumDivideTime;
+    uint public lastTimeQuorumReached;
+
+    uint public votingTime;
+    uint public currentVotingTime;
+
+    string constant voteTokenName = "Vote Token";
+    string constant quorumTokenName = "Quorum Token";
+    string constant voteTokenSymbol = "VOTE";
+    string constant quorumTokenSymbol = "QUORUM";
+    uint8 constant DECIMALS = 18;
+    address public constant supportDeposit = 0x707574546F566F74650000000000000000000000; // When this address reaches quorum proposal gets put to vote.
+    address public constant approvalDeposit =  0x617070726f76616c000000000000000000000000; // This address represents yes votes.
+    address public constant rejectionDeposit = 0x72656a656374696F6E0000000000000000000000; // This address represents no votes.
+
+    enum ProposalState {
+        New,
+        PutToSupport,
+        PutToVote,
+        Decided
+    }
+
+    struct Proposal {
+        address destination; // The governed contract.
+        uint amount; // The amount of ETH to send (in most cases it should be 0).
+        bytes data; // The data (similar to a multisig mechanism).
+        string uriDescription; // URI to a natural language description of the proposal. Also used as ID in mappings.
+        bytes32 descriptionHash; // Hash of the description.
+        string uriArguments; // URI to arguments for the proposal.
+        bytes32 argumentsHash; // Hash of the arguments.
+        ProposalState state; // State of proposal.
+        uint whenPutToVote; // Records the time when a proposal put to vote to be able to calculate voting period.
+        MiniMeToken quorumToken; // The token that will be used for quorum.
+        MiniMeToken voteToken; // The token that will be used for actual proposal voting.
+        bool approved; // Outcome of voting.
+    }
+
+    mapping(bytes32 => Proposal) public proposals;
+
+
+    constructor (uint _proposalQuorum, uint _quorumDivideTime, uint _votingTime, ArbitrablePermissionList _arbitrablePermissionList, Pinakion _pinakion, TokenController _tokenController) public {
+        lastTimeQuorumReached = block.timestamp;
+
+        proposalList = _arbitrablePermissionList;
+        pinakion = _pinakion;
+        tokenController = _tokenController;
+
+        proposalQuorum = _proposalQuorum;
+        currentProposalQuorum = proposalQuorum;
+
+        quorumDivideTime = _quorumDivideTime;
+        votingTime = _votingTime;
+    }
+
+
+    // ****************************** //
+    // *          Modifiers         * //
+    // ****************************** //
+
+    modifier onlyItself() {
```

Comment:
Why not just use `internal` or `private`?

Reply:
Correct me if I'm wrong:

When internal, a contract which inherits Governance (which is unauthorized) contract can call it.
When private, a contract which inherits Governance contract won't be able to reach that function, has to redefine.

Reply:
So what is wrong with private? If the inheriting contract is unauthorized to call it, why does it need to be able to reach it?

Reply:
As far as I know subclasses do not inherit private members, so a subclass will have to define those functions again. Shortly, extra work for users of this class.

Reply:
No, you are right, `internal` does the job, `onlyItself` is unnecessary. Gonna fix this tomorrow. Thanks.